src/hooks/useListSettings.ts

```diff
@@ -12,6 +12,13 @@ export interface UseListSettings<TColumns extends string = string> {
     value: ListSettings<TColumns>[T],
   ) => void;
 }
+
+const mergeCustomizer = (objValue: unknown, srcValue: unknown) => {
+  if (Array.isArray(objValue) && Array.isArray(srcValue)) {
+    return srcValue;
+  }
+};
+
```

Comment:
We have a unit test for this hook already written - it passes, problems could only occur when deeply-nested array values are added to `listConfig` in the future, but I believe this will be detected easily. I've also added a comment.